### ISSUE TITLE
FOQUS pip install Launch Error

### ISSUE THREAD

**Issue by bpaul4** [CCSI-Toolset/FOQUS#928]
This error concerns the local user install of CCSI-FOQUS v3.9.0 in a Python 3.8.10 Miniconda environment. Per the documentation instructions (Read The Docs > FOQUS Installation and Running > Quick Start instructions), 'pip install foqus' runs smoothly and 'foqus' launch command fails due to missing 'pywin32' dependency which must be manually installed ('pip install pywin32') to launch FOQUS.

**Comment by ksbeattie** [comment]
Could you try this again with `pip install ccsi-foqus` command?  `foqus` is, unfortunately, an entirely different product.

**Comment by bpaul4** [comment]
On my personal machine, running 'pip install pywin32' fixed the issue immediately (it did not when I tried installing/launching on my machine with the GitHub installation, not sure what changed). The installed packages include ccsi-foqus and pywin32 on the conda package list.

**Comment by bpaul4** [comment]
After installing ccsi-foqus I received this traceback:
(ccsi-foqus) C:\Users\bpaul>foqus
Traceback (most recent call last):
  File "c:\users\bpaul\miniconda3\envs\ccsi-foqus\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\bpaul\miniconda3\envs\ccsi-foqus\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\bpaul\Miniconda3\envs\ccsi-foqus\Scripts\foqus.exe\__main__.py", line 4, in <module>
  File "c:\users\bpaul\miniconda3\envs\ccsi-foqus\lib\site-packages\foqus_lib\foqus.py", line 20, in <module>
    from foqus_lib.framework.session.session import *
  File "c:\users\bpaul\miniconda3\envs\ccsi-foqus\lib\site-packages\foqus_lib\framework\session\session.py", line 17, in <module>
    from foqus_lib.framework.graph.graph import *
  File "c:\users\bpaul\miniconda3\envs\ccsi-foqus\lib\site-packages\foqus_lib\framework\graph\graph.py", line 23, in <module>
    from foqus_lib.framework.graph.node import *  # Node, input var and output var classes
  File "c:\users\bpaul\miniconda3\envs\ccsi-foqus\lib\site-packages\foqus_lib\framework\graph\node.py", line 20, in <module>
    from foqus_lib.framework.sim.turbineConfiguration import TurbineInterfaceEx
  File "c:\users\bpaul\miniconda3\envs\ccsi-foqus\lib\site-packages\foqus_lib\framework\sim\turbineConfiguration.py", line 32, in <module>
    import win32process
ImportError: DLL load failed while importing win32process: The specified module could not be found.

**Comment by bpaul4** [comment]
`conda install pywin32` seemed to fix the error and `foqus` launches correctly now. `pywin32` was installed from the `ccsi-foqus` installation but seems not to have linked to the environment until manually installed.

**Comment by ksbeattie** [comment]
I can't reproduce this.  Could you try again, from a completely new conda environment?

I suspect that the problem is when you ran `pip install foqus` (which is not in the instructions) rather than `pip install ccsi-foqus` (which is in the instructions).  But even trying that, I couldn't reproduce it.

**Comment by bpaul4** [comment]
I recreated the installation workflow by running `conda create --name test-ccsi-foqus python=3.8`, then `conda activate test-ccsi-foqus` and installed with `pip install ccsi-foqus`. Attempting to create a desktop shortcut via `foqus --make-shortcut` or launch via `foqus` reproduced the win32 process DLL import error

**Comment by lbianchi-lbl** [comment]
I think this might be due to a known issue when the PyPI version of `pywin32` is installed in a Conda environment using Python version 3.8 or greater, see e.g. https://github.com/IDAES/idaes-pse/issues/258. If the environment is created with Python 3.7 or lower, as shown in the installation docs, the issue would not occur. We're currently working towards making FOQUS fully compatible with Python 3.8 (see #918), which should also include test cases verifying compatibility issues such as this one.

In summary: although strictly speaking this does not apply to the current installation instructions (at least when followed *verbatim*), I think this is a real issue that we should address as part of #918. I'll go ahead and add there the usual fix/workaround to address the version incompatibility.

**Comment by ksbeattie** [comment]
Ah yes, that was it!  I was able to reproduce it when trying with `python=3.8` and yes this underscores that:
- we don't yet support python 3.8 and
- we need to

**Comment by bpaul4** [comment]
Okay, thank you for clarifying the issue. The Anaconda install links in both the CCSI and IDAES docs redirect to the Individual Edition page for version 3.8 download. It appears older versions are archived, and Miniconda for Python 3.7 is offered only for Linux, not Windows or MacOSX (2.7, 3.8 and 3.9 are offered for all 3 systems).

**Comment by lbianchi-lbl** [comment]
@bpaul4 Note that the Python version mentioned in the Miniconda/Anaconda download page only determines the *default* Python version of Conda environments, i.e. which Python version will be used if no version for the `python` Conda package is specified when creating the environment.

It's always possible to create Conda environments with arbitrary Python versions (regardless of the default Python version that was used when installing Anaconda/Miniconda) as long as it is specified, e.g.:

```sh
conda create -n my-env python=3.7
```

This is mentioned e.g. in (a very well hidden section of) the Miniconda installation page [here](https://docs.conda.io/en/latest/miniconda.html#other-resources). Although none of this is specific to FOQUS in particular, I fully appreciate that this might not be immediately evident from the docs. In any case, the fact that Python 3.8 is currently the default version for Conda was our original and primary motivation for our ongoing effort in #918 to make sure that FOQUS is fully compatible with Python 3.8.

**Comment by ksbeattie** [comment]
While this is still possible with the 3.10.0 release, now that #918 has been merged in, it will be fixed for the next release.
